src/spyglass/lfp/v1/lfp.py

```diff
@@ -260,6 +264,287 @@ def get_lfp_object(key: dict):
             return ImportedLFP & (LFPOutput.ImportedLFP & key).fetch("KEY")
 
 
+# add artifact detection
+@schema
+class LFPArtifactDetectionParameters(dj.Manual):
+    definition = """
+    # Parameters for detecting LFP artifact times within a LFP group.
+    artifact_params_name: varchar(200)
+    ---
+    artifact_params: blob  # dictionary of parameters
+    """
+
+    def insert_default(self):
+        """Insert the default artifact parameters with an appropriate parameter dict."""
+        artifact_params = {}
+        artifact_params["zscore_thresh"] = None  # must be None or >= 0
+        artifact_params["amplitude_thresh"] = 500  # must be None or >= 0, ad units
+        # all electrodes of sort group
+        artifact_params["proportion_above_thresh"] = 0.1
+        artifact_params["removal_window_ms"] = 3.0  # in milliseconds
+        self.insert1(["default", artifact_params], skip_duplicates=True)
+
+        artifact_params_none = {}
+        artifact_params_none["zscore_thresh"] = None
+        artifact_params_none["amplitude_thresh"] = None
+        self.insert1(["none", artifact_params_none], skip_duplicates=True)
+
+
+@schema
+class LFPArtifactDetectionSelection(dj.Manual):
+    definition = """
+    # Specifies artifact detection parameters to apply to a sort group's recording.
+    -> LFP
+    -> LFPArtifactDetectionParameters
+    ---
+    custom_artifact_detection=0 : tinyint
```

Comment:
What is custom artifact detection?

Reply:
Can you either add the z-score detection in or take it out of the parameters. Otherwise we just have code that doesn't do anything.